presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -2355,17 +2376,33 @@ private static Domain buildColumnDomain(ColumnHandle column, List<HivePartition>
             throw new PrestoException(NOT_SUPPORTED, "Writing to bucketed sorted Hive tables is disabled");
         }
 
+        HivePartitioningHandle hivePartitioningHandle;
         List<String> bucketedBy = bucketProperty.get().getBucketedBy();
-        Map<String, HiveType> hiveTypeMap = tableMetadata.getColumns().stream()
-                .collect(toMap(ColumnMetadata::getName, column -> toHiveType(typeTranslator, column.getType())));
-        return Optional.of(new ConnectorNewTableLayout(
-                new HivePartitioningHandle(
+
+        HiveBucketProperty.BucketingType bucketingType = bucketProperty.get().getBucketingType();
```

Comment:
We never directly create temporary table. We always insert. Why the change to the `getNewTableLayout` is needed?

Reply:
Then this is not needed, will remove.